lib/background-script.js

```diff
@@ -0,0 +1,63 @@
+const defaultConfig = {
+  blacklist: [
+    'facebook.com',
+    'www.facebook.com',
+    'messenger.com',
+    'www.messenger.com',
+    'twitter.com',
+    'www.twitter.com',
+    'www.youtube.com',
+    'youtube.com',
+    'dailymotion.com',
+    'www.dailymotion',
+    'vimeo.com',
+    'www.vimeo.com',
+    'instagram.com',
+    'www.instagram.com',
+    'periscope.com',
+    'www.periscope.com',
+    'linkedin.com',
+    'www.linkedin.com',
+    'viadeo.com',
+    'www.viadeo.com',
+    'plus.google.com',
+    'tumblr.com',
+    'www.tumblr',
+    'pinterest.com',
+    'www.pinterest.com',
+    'reddit.com',
+    'www.reddit.com',
+    'www.vk.com',
+    'vk.com',
+    'mastodon.qowala.org'
+  ]
+}
+
+window.setInterval(() => {
+  browser.storage.local.get({
```

Comment:
To keep the plugin compatible with the Chrome browser, you need to use the `chrome.storage` function. It will work on Firefox too, but the `browser.storage` method doesn't work on Chrome (I tested on Chromium `58.0.3029.110`).